rs/replicated_state/src/canister_state/queues/queue.rs

```diff
@@ -332,25 +295,28 @@ impl TryFrom<(pb_queues::CanisterQueue, Context)> for CanisterQueue {
     type Error = ProxyDecodeError;
 
     fn try_from((item, context): (pb_queues::CanisterQueue, Context)) -> Result<Self, Self::Error> {
-        let queue: VecDeque<CanisterQueueItem> = item
+        let queue: VecDeque<message_pool::Id> = item
             .queue
             .into_iter()
             .map(|queue_item| match queue_item.r {
                 Some(pb_queues::canister_queue::queue_item::R::Reference(_)) => {
-                    let id = message_pool::Id::try_from(queue_item)?;
-                    if id.context() != context {
+                    let reference = message_pool::Id::try_from(queue_item)?;
```

Comment:
Not this change, but it seems we are skipping over stale entries here. Couldn't this lead to non-determinism when a replica restarts and loads freshly from the checkpoint whereas others continued from the in-memory checkpoint?

Reply:
No, this is simply about whether we're able to decode a `message_pool::Id` from the protobuf. (Which [may fail](https://sourcegraph.com/github.com/dfinity/ic@20321d1029aa2d0dff8c9148b1c236f5d7a78713/-/blob/rs/replicated_state/src/canister_state/queues/message_pool.rs?L137&symbol=rust-analyzer+cargo+ic-replicated-state+0.9.0+message_pool/queues/canister_state/Id%23TryFrom%23try_from%28%29.) if the `oneof` field is `None`.)

Reply:
Ohhhh... 🤦 sorry for the noise it is too early...